### PR TITLE
:broom: refurb cleanup + one typo

### PR DESCRIPTION
**Pull request by jayceslesar**
Used the python library [refurb](https://github.com/dosisod/refurb) against this repo and added most of the suggested changes.

Feel free to counteract any!

I think good future work will be dropping support for anything lower than python 3.9 as well and porting all annotations to future syntax in terms of codebase wide changes.

### CODE REVIEW

**Comment by jonasvdd** [comment]
Hi @jayceslesar! Really cool repo! Had not heard from it yet!

Based on [these statistics](https://w3techs.com/technologies/history_details/pl-python/3), `python 3.8` still is on the rise.
However, as the older versions would still be supported by e.g., `plotly-resampler<=0.8.X`, there is a case  for being more severe with our version support.

What would be the main advantage of porting these annotations @jayceslesar?
What are your thoughts on this @jvdd.

I will review your code somewhere this week! 

And of course, 
Happy holidays! 🎅🏼

File: plotly_resampler/figure_resampler/figure_resampler_interface.py [hunk 1]
```diff
@@ -1291,14 +1291,14 @@ def construct_update_data(
...
-            elif len(autorange_matches) and not len(spike_matches):
+            elif autorange_matches and not spike_matches:
                 # PreventUpdate returns a 204 status code response on the
                 # relayout post request
                 return dash.no_update
 
         # If we do not have any traces to be updated, we will return an empty
         # request response
-        if len(updated_trace_indices) == 0:
+        if not updated_trace_indices:
```
Comment (jonasvdd) [inline plotly_resampler/figure_resampler/figure_resampler_interface.py:1301]: This is for me somewhat less verbose / clear than the `len(...) == 0`

Comment (jonasvdd) [reply plotly_resampler/figure_resampler/figure_resampler_interface.py:1301]: What is you opinion on this @jvdd?

File: plotly_resampler/figure_resampler/utils.py [hunk 2]
```diff
@@ -128,7 +128,7 @@ def timedelta_to_str(td: pd.Timedelta) -> str:
     if c.days > 0:
         out_str += f"{c.days}D"
     if c.hours > 0 or c.minutes > 0 or c.seconds > 0 or c.milliseconds > 0:
-        out_str += "_" if len(out_str) else ""
+        out_str += "_" if out_str else out_str
```
Comment (jonasvdd) [inline plotly_resampler/figure_resampler/utils.py:131]: Again, this is here somewhat less clear for me (might be personal flavor).
e.g., I would at lease propose to retain the  `else ""`  part.

**Review by jonasvdd — COMMENTED** [review]
LGTM!

Certainly an improvement of the code quality! 👏🏼 
Can you give some feedback @jvdd?

File: plotly_resampler/figure_resampler/utils.py (hunk 2 above)
Comment (jayceslesar) [reply plotly_resampler/figure_resampler/utils.py:131]: I see your point there -- some of the refurb suggestions I also disagree with! Will make this change

Comment (jayceslesar) [inline plotly_resampler/figure_resampler/utils.py:131]:
```suggestion
        out_str += "_" if out_str else ""
```

**Comment by jayceslesar** [comment]
> Hi @jayceslesar! Really cool repo! Had not heard from it yet!
> 
> Based on [these statistics](https://w3techs.com/technologies/history_details/pl-python/3), `python 3.8` is still on the rise. However, as the older versions would still be supported by e.g., `plotly-resampler<=0.8.X`, there is a case for being more severe with our version support.
> 
> What would be the main advantage of porting these annotations @jayceslesar? What are your thoughts on this @jvdd.
> 
> I will review your code somewhere this week!
> 
> And of course, Happy holidays! 🎅🏼

I guess I am biased that the new annotations are really just nicer to look at haha, but I see your points, no reason to not support older versions if they are still being widely used

And cheers! Happy holidays

File: plotly_resampler/figure_resampler/figure_resampler_interface.py (hunk 1 above)
Comment (jvdd) [reply plotly_resampler/figure_resampler/figure_resampler_interface.py:1301]: Is indeed less verbose, but I guess this adjustment is more "Pythonic".
We can make it perhaps more verbose by adding a small inline comment?

File: plotly_resampler/figure_resampler/figure_resampler_interface.py [hunk 3]
```diff
@@ -1336,7 +1336,7 @@ def _parse_dtype_orjson(series: np.ndarray) -> np.ndarray:
         #    * float16 and float128 aren't supported with latest orjson versions (3.8.1)
         #    * this method assumes that the it will not get a float128 series
         # -> this method can be removed if orjson supports float16
-        if series.dtype in [np.float16]:
+        if series.dtype in (np.float16):
```
Comment (jvdd) [inline plotly_resampler/figure_resampler/figure_resampler_interface.py:1339]: This line causes the test to fail, as this is not a tuple (& thus not iterable) but just a dtype with parentheses around it.

File: plotly_resampler/figure_resampler/figure_resampler_interface.py (hunk 1 above)
Comment (jvdd) [reply plotly_resampler/figure_resampler/figure_resampler_interface.py:1301]: imo, the surrounding comments make it quite clear that updated_trace_indices should be empty for this if to pass

**Review by jvdd — APPROVED** [review]
LGTM! Applied some minor changes myself :)

**Comment by jvdd** [comment]
I think merging this PR will make you (@jayceslesar) the first contributor besides @jonasvdd & me to the Python code :tada:

**Review by jonasvdd — APPROVED** [review]

**Comment by jonasvdd** [comment]
Thanks a lot @jayceslesar! I'll merge this one!
 🤝

**Pull request merged**